src/partition/ParquetInterface.cc

```diff
@@ -0,0 +1,353 @@
+        m_recordBufferSize = GetStringRecordSize(schema);
+        // batchSize_buf = int((maxBufferSize*1024*1024)/m_recordBufferSize);
+        batchSize_buf = int(maxBufferSize / m_recordBufferSize);
+        std::cout << "\nMax buffer size : " << maxBufferSize << " vs " << m_recordBufferSize
+                  << "  -> batch size : " << batchSize_buf << std::endl;
+    }
+
+    m_batchSize = std::min(batchSize_mem, batchSize_buf);
+    m_arrow_reader_gbl->set_batch_size(m_batchSize);
+    m_totalBatchNumber = int(fileRowNumber / m_batchSize);
+    if (m_totalBatchNumber * m_batchSize < fileRowNumber) m_totalBatchNumber++;
+
+    std::cout << "Number of rows : " << fileRowNumber << "  batchSize " << m_batchSize << std::endl;
+    std::cout << "RecordBatchReader : batch number " << m_totalBatchNumber << std::endl;
+    return arrow::Status::OK();
+}
+
+int ParquetFile::GetTotalRowNumber(std::string fileName) const {
+    std::shared_ptr<arrow::io::ReadableFile> infile;
+    PARQUET_ASSIGN_OR_THROW(infile, arrow::io::ReadableFile::Open(fileName, arrow::default_memory_pool()));
+
+    std::unique_ptr<parquet::arrow::FileReader> reader;
+    PARQUET_THROW_NOT_OK(parquet::arrow::OpenFile(infile, arrow::default_memory_pool(), &reader));
+
+    std::shared_ptr<parquet::FileMetaData> metadata = reader->parquet_reader()->metadata();
+    // std::cout<<"Metadata version : "<<metadata->version()<<std::endl;
+    // std::cout<<"Metadata created : "<<metadata->created_by() <<std::endl;
+    // std::cout<<"Metadata #row groups : "<<metadata->num_row_groups() <<std::endl;
+    // std::cout<<"Metadata #rows : "<<metadata->num_rows() <<std::endl;
+
+    return metadata->num_rows();
+}
+
+// Read an arrow batch, format the table acording to the config file and save it in csv format
+arrow::Status ParquetFile::ReadNextBatch_Table2CSV(void* buf, int& buffSize,
+                                                   std::vector<std::string> params) {
+    std::shared_ptr<arrow::Table> table_loc;
+
+    m_parameterNames = params;
+    arrow::Status batchStatus = ReadNextBatchTable_Formatted(table_loc);
+
+    if (!batchStatus.ok()) return arrow::Status::ExecutionError("Error while reading and formating batch");
+
+    arrow::Status status = Table2CSVBuffer(table_loc, buffSize, buf);
+
+    if (status.ok()) return arrow::Status::OK();
+
+    return arrow::Status::ExecutionError("Error while writing table to CSV buffer");
+}
+
+// Create char buffer containing the table in csv format
+arrow::Status ParquetFile::Table2CSVBuffer(std::shared_ptr<arrow::Table>& table, int& buffSize, void* buf) {
+    ARROW_ASSIGN_OR_RAISE(auto outstream, arrow::io::BufferOutputStream::Create(1 << 10));
+
+    // Options : null string, no header, no quotes around strings
+    arrow::csv::WriteOptions writeOpt = arrow::csv::WriteOptions::Defaults();
+    writeOpt.null_string = "\\N";
+    writeOpt.include_header = false;
+    writeOpt.quoting_style = arrow::csv::QuotingStyle::None;
```

Comment:
I'm not sure that the hard-coding parameters of the `CSV` dialect for the output stream should be done here. The dialect is specified in the configuration object (files) of the partitioning tool. You may need to use those options from there and pass those into `arrow::csv::WriteOptions`.

Reply:
Done by replacing the paramNames string vector by a struct object containing also these csv formatting strings.